src/System.Net.Http/src/System/Net/Http/Headers/HttpContentHeaders.cs

```diff
@@ -148,5 +148,21 @@ internal HttpContentHeaders(HttpContent parent)
         {
             _parent = parent;
         }
+
+        internal override void EnsureHeaders()
+        {
+            base.EnsureHeaders();
+            if (_parent != null)
+            {
+                // Content-Length is not among headers unless its property is accessed.
+                long? contentLengthValue = ContentLength;
+            }
+        }
+
+        public override string ToString()
```

Comment:
This adds a new override to a public method `ToString()`.  That will have to be added to the 'ref' folder as an API change.

cc: @stephentoub

Reply:
While ideally it would be, technically it doesn't have to, as the only impact this would have is on a derived type, and this type is sealed.